connector/src/main/java/com/datastax/oss/pulsar/source/converters/NativeJsonConverter.java

```diff
@@ -0,0 +1,407 @@
+            case ProtocolConstants.DataType.DOUBLE:
+                return jsonNodeFactory.numberNode(record.getDouble(identifier));
+            case ProtocolConstants.DataType.FLOAT:
+                return jsonNodeFactory.numberNode(record.getFloat(identifier));
+            case ProtocolConstants.DataType.BOOLEAN:
+                return jsonNodeFactory.booleanNode(record.getBoolean(identifier));
+            case ProtocolConstants.DataType.TIMESTAMP:
+                return jsonNodeFactory.numberNode(record.getInstant(identifier).toEpochMilli());
+            case ProtocolConstants.DataType.DATE: // Mimic Avro date (epoch days)
+                return jsonNodeFactory.numberNode((int) record.getLocalDate(identifier).toEpochDay());
+            case ProtocolConstants.DataType.TIME: // Mimic Avro time (microseconds)
+                return jsonNodeFactory.numberNode (record.getLocalTime(identifier).toNanoOfDay() / 1000);
+            case ProtocolConstants.DataType.BLOB:
+                return jsonNodeFactory.binaryNode(getBytes(record.getByteBuffer(identifier)));
+            case ProtocolConstants.DataType.UDT:
+                return buildUDTValue(record.getUdtValue(identifier));
+            case ProtocolConstants.DataType.DURATION:
+                return cqlDurationToJsonNode(record.getCqlDuration(identifier));
+            case ProtocolConstants.DataType.DECIMAL:
+                return bigDecimalToJsonNode(record.getBigDecimal(identifier));
+            case ProtocolConstants.DataType.VARINT:
+                Conversion<BigInteger> conversion = SpecificData.get().getConversionByClass(BigInteger.class);
+                ByteBuffer bf = conversion.toBytes(record.getBigInteger(identifier), CqlLogicalTypes.varintType, CqlLogicalTypes.CQL_VARINT_LOGICAL_TYPE);
+                return jsonNodeFactory.binaryNode(getBytes(bf));
+            case ProtocolConstants.DataType.LIST: {
+                ListType listType = (ListType) dataType;
+                String path = getSubSchemaPath(record, identifier);
+                org.apache.avro.Schema listSchema = subSchemas.get(path);
+                List listValue = record.getList(identifier, CodecRegistry.DEFAULT.codecFor(listType.getElementType()).getJavaType().getRawType());
+                log.info("field={} listSchema={} listValue={}", identifier, listSchema, listValue);
+                return createArrayNode(listSchema, listValue);
+            }
+            case ProtocolConstants.DataType.SET: {
+                SetType setType = (SetType) dataType;
+                String path = getSubSchemaPath(record, identifier);
+                org.apache.avro.Schema setSchema = subSchemas.get(path);
+                Set setValue = record.getSet(identifier, CodecRegistry.DEFAULT.codecFor(setType.getElementType()).getJavaType().getRawType());
+                log.debug("field={} setSchema={} setValue={}", identifier, setSchema, setValue);
```

Comment:
I would add
if log.isDebugEnabled()

the are two schools of thought about this.
My personal position is that if the code style of the project uses the guards, it is better to use them everywhere (like we do in Pulsar, BookKeeper and ZooKeeper for instance).
If the overall project style is to not use the guards, it is better to not use them.

I personally tend to prefer the guards and to not rely too much on JVM optimisations for log statements

Reply:
this project style doesn't use guards. I can add them everywhere if the need arise outside of this patch.